contracts/src/v0.8/ccip/pools/GHO/UpgradeableBurnMintTokenPool.sol

```diff
@@ -0,0 +1,78 @@
+import {IBurnMintERC20} from "../../../shared/token/ERC20/IBurnMintERC20.sol";
+
+import {UpgradeableBurnMintTokenPoolAbstract} from "./UpgradeableBurnMintTokenPoolAbstract.sol";
+import {UpgradeableTokenPool} from "./UpgradeableTokenPool.sol";
+
+import {Initializable} from "solidity-utils/contracts/transparent-proxy/Initializable.sol";
+import {IRouter} from "../../interfaces/IRouter.sol";
+
+/// @title UpgradeableBurnMintTokenPool
+/// @author Aave Labs
+/// @notice Upgradeable version of Chainlink's CCIP BurnMintTokenPool
+/// @dev Contract adaptations:
+/// - Implementation of Initializable to allow upgrades
+/// - Move of allowlist and router definition to initialization stage
+/// - Add GHO-Specific onlyOwner `directMint` which mints liquidity to the old pool &
+/// increases facilitator level.
+/// - Add GHO-Specific onlyOwner `directBurn` which burns liquidity & reduces facilitator level.
+/// - Remove i_token decimal check in UpgradeableTokenPool constructor
+
+/// @dev Pool whitelisting mode is set in the constructor and cannot be modified later.
+/// It either accepts any address as originalSender, or only accepts whitelisted originalSender.
+/// The only way to change whitelisting mode is to deploy a new pool.
+/// If that is expected, please make sure the token's burner/minter roles are adjustable.
+/// @dev This contract is a variant of BurnMintTokenPool that uses `burn(amount)`.
+contract UpgradeableBurnMintTokenPool is Initializable, UpgradeableBurnMintTokenPoolAbstract, ITypeAndVersion {
+  string public constant override typeAndVersion = "BurnMintTokenPool 1.5.1";
+
+  constructor(
+    address token,
+    uint8 localTokenDecimals,
+    address rmnProxy,
+    bool allowListEnabled
+  ) UpgradeableTokenPool(IBurnMintERC20(token), localTokenDecimals, rmnProxy, allowListEnabled) {}
+
+  function initialize(address owner_, address[] memory allowlist, address router) external initializer {
+    if (router == address(0) || owner_ == address(0)) revert ZeroAddressNotAllowed();
+
+    _transferOwnership(owner_);
+    s_router = IRouter(router);
+    if (i_allowlistEnabled) _applyAllowListUpdates(new address[](0), allowlist);
+  }
+
+  /// @notice This function allows the owner to mint `amount` tokens on behalf of the pool
+  /// and transfer them to `to`. This is GHO-Specific and is called to match the facilitator
+  /// level of the new pool with the old pool such that it can burn the bridged supply once
+  /// the old pool is deprecated. The old pool is then expected to burn `amount` of tokens
+  /// so that it can be removed as a facilitator on GHO (ideally using `directBurn`).
+  /// @dev This is only called while offboarding an old token pool (or facilitator) in favor
+  /// of this pool.
+  /// @param to The address to which the minted tokens will be transferred. This needs to
+  /// be the old token pool,
+  /// or the facilitator being offboarded.
+  /// @param amount The amount of tokens to mint and transfer to old pool.
```

Comment:
I think it's better if we keep it more generic as well as giving insights about the pool migration process. wdyt
```suggestion
  /// @notice This function allows the owner to mint `amount` tokens on behalf of the pool
  /// and transfer them to `to`. 
  /// @dev This GHO-specific functionality is designed for migrating bucket levels between facilitators. The old 
  /// pool is expected to burn amount of tokens, allowing this pool to mint an equivalent amount.
  /// @param to The address to which the minted tokens will be transferred.
  /// @param amount The amount of tokens to mint and transfer.
```

Reply:
```The old pool is expected to burn `amount` of tokens, allowing this pool to mint an equivalent `amount`.```
i'm not sure about this line, i think more accurate phrasing would be: ```The new pool is expected to mint `amount` of tokens, and transfer to the old pool allowing it the burn an equivalent amount. 
This increases the facilitator level of the new pool, allowing it offramp existing supply on the token: since a facilitator can only burn (during offramp) an amount is it has previously minted.```
wdyt?

Reply:
[4e0e90f](https://github.com/aave/ccip/pull/20/commits/4e0e90fbded84e592affeea4dacf5e597f6a251b)